social_core/backends/mediawiki.py

```diff
@@ -0,0 +1,175 @@
+"""
+MediaWiki OAuth1 backend
+========================
+
+Usage
+-----
+
+In addition to the general setup you need to define the
+following parameters. In Django's settings.py you would
+use for English Wikipedia::
+
+    SOCIAL_AUTH_MEDIAWIKI_KEY = <consumer_key>
+    SOCIAL_AUTH_MEDIAWIKI_SECRET = <consumer_secret>
+    REQUEST_TOKEN_URL =
+        'https://en.wikipedia.org/w/index.php?title=Special:OAuth/initiate'
+    AUTHORIZATION_URL =
+        'https://en.wikipedia.org/w/index.php?title=Special:Oauth/authorize'
+    ACCESS_TOKEN_URL =
+        'https://en.wikipedia.org/w/index.php?title=Special:Oauth/token'
+    IDENTIFY_URL =
+        'https://en.wikipedia.org/w/index.php'
+
+If users should only be prompeted once for permission use:
+
+    AUTHORIZATION_URL =
+        'https://en.wikipedia.org/w/index.php?title=Special:Oauth/authenticate'
+
+General documentation
+---------------------
+
+https://www.mediawiki.org/wiki/Extension:OAuth
+
+Developer documentation
+-----------------------
+
+https://www.mediawiki.org/wiki/OAuth/For_Developers
+"""
+
+import jwt
+import requests
+import six
+from requests_oauthlib import OAuth1
+from six import b
+from six.moves.urllib.parse import parse_qs, urlencode
+from .oauth import BaseOAuth1
+
+
+class MediaWiki(BaseOAuth1):
+
+    """
+    Handles the handshake with Mediawiki and fetching of user data.
+    """
+
+    name = 'mediawiki'
+    REQUEST_TOKEN_URL = 'https://en.wikipedia.org/w/index.php' + \
+                        '?title=Special:OAuth/initiate'
+    AUTHORIZATION_URL = 'https://en.wikipedia.org/w/index.php' + \
+                        '?title=Special:Oauth/authorize'
+    ACCESS_TOKEN_URL = 'https://en.wikipedia.org/w/index.php' + \
+                       '?title=Special:Oauth/token'
+    IDENTIFY_URL = 'https://en.wikipedia.org/w/index.php' + \
+                   '?title=Special:OAuth/identify'
+
+    def unauthorized_token(self):
+        """
+        Return request for unauthorized token (first stage)
+
+        Mediawiki request token is requested from e.g.:
+         * https://en.wikipedia.org/w/index.php?title=Special:OAuth/initiate
+         * 'callback_uri' needs to be 'oob'
```

Comment:
I added the option to set the callback in the settings and updated the docs. Thanks for the explanation. I had some trouble understanding that before.